src/App.tsx

```diff
@@ -1,222 +1,203 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import React, {
+  useCallback,
+  useContext,
+  useEffect,
+  // useMemo,
+  useRef,
+  useState,
+} from 'react';
+import cn from 'classnames';
+import {
+  getTodos,
+  createTodo,
+  removeTodo,
+  changeTodo,
+} from './api/todos';
 import { AuthContext } from './components/Auth/AuthContext';
+import { Todo } from './types/Todo';
+import { FilterCondition, ErorrMessage } from './types/enums';
+import { Header } from './components/Main/Header';
+import { TodoList } from './components/Main/TodoList';
+import { Footer } from './components/Main/Footer';
 
 export const App: React.FC = () => {
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const user = useContext(AuthContext);
   const newTodoField = useRef<HTMLInputElement>(null);
 
+  const [todosList, setTodosList] = useState<Todo[]>([]);
+  const [isError, setIsError] = useState(false);
+  const [errorText, setErrorText] = useState('');
+  const [filterCondition, setFilterCondition] = useState(FilterCondition.ALL);
+  const [isAdding, setIsAdding] = useState(false);
+  const [tempNewTask, setTempNewTask] = useState<Todo | null>(null);
+  const [deletingTodoIds, setDeletingTodoIds] = useState<number[]>([]);
+  const [updatingTodoIds, setUpdatingTodoIds] = useState<number[]>([]);
+
+  if (isError) {
+    setTimeout(() => setIsError(false), 3000);
+  }
```

Comment:
Create `showError` function and move this logic into there

Reply:
+